src/project/mod.rs

```diff
@@ -0,0 +1,60 @@
+//! Project management module
+//!
+//! Manages configuration files, paths, and such.
+
+use std::io;
+use std::path::{Path, PathBuf};
+
+/// Configuration macros.
+#[macro_use]
+pub mod config;
+/// Directory and file loading.
+pub mod directory;
+
+pub use self::config::Config;
+
+/// Error related to anything that manages/creates configurations as well as
+/// "workspace"-related things.
+#[derive(Debug)]
+pub enum ProjectError {
+    /// Forward to the `std::io::Error` error.
+    File(io::Error),
+    /// Errors related to serde's parsing of configuration files.
+    Parser(String),
+}
+
+impl From<io::Error> for ProjectError {
+    fn from(e: io::Error) -> ProjectError {
+        ProjectError::File(e)
+    }
+}
+
+impl ProjectError {
+    /// Returns a human friendly error message for the `ProjectError`.
+    pub fn description(&self) -> String {
+        match self {
+            &ProjectError::File(ref err) => err.to_string(),
+            &ProjectError::Parser(ref msg) => msg.clone(),
+        }
+    }
+}
+
+/// Project structure, holds information related to configurations and meta information.
+pub struct Project {
```

Comment:
I was hoping to eventually use it as some sort of meta-ish structure about the project, such as defining paths to resources, configurations, assets, etc. It isn't using `Directory` right now because I'm looking at using the walkdir crate instead, but I'm not 100% sure either way.

`#[derive(Debug)]` can be added I suppose, this is just a little mockup for what I want to do with this though, the main part of this PR is just dealing with configurations.